### PR TITLE
Refactor/ADF-213/Allow providing controller to check action access

### PR DESCRIPTION
**Pull request by shpran**
**Relates to:** _[ADF-213](https://oat-sa.atlassian.net/browse/ADF-213)_

### CODE REVIEW

File: actions/class.CommonModule.php [hunk 1]
```diff
@@ -106,9 +106,9 @@ protected function hasAccess($controllerClass, $action, $parameters = [])
         return AclProxy::hasAccess($user, $controllerClass, $action, $parameters);
     }
 
-    protected function hasWriteAccessToAction(string $action, ?User $user = null): bool
+    protected function hasWriteAccessToAction(string $action, ?string $controller = null, ?User $user = null): bool
```
Comment (gabrielfs7) [inline actions/class.CommonModule.php:109]: Is there a way to avoid a breaking change?

Comment (shpran) [reply actions/class.CommonModule.php:109]: #### Solution 1: Change the `$action` property to `$controllerAction` and allow values such as `Controller::action` to be provided.
```php
/**
 * @example        $this->hasWriteAccessToAction(__FUNCTION__);
 * @example        $this->hasWriteAccessToAction(MediaManager::class . '::' . __FUNCTION__)
 */
public function hasWriteAccessToAction(string $controllerAction, ?User $user): bool
{
    $controllerActionParts = explode('::', $controllerAction);

    $controller = count($controllerActionParts) === 2
        ? $controllerActionParts[0]
        : static::class;
    $action = $controllerActionParts[1] ?? $controllerActionParts[0];
    
    return $this->getActionAccessControl()->hasWriteAccess($controller, $action, $user);
}
```
---
#### Solution 2: mark the old one as deprecated and create a new method that will expose the controller.
```php
/**
 * @deprecated Use hasWriteAccessToControllerAction
 */
protected function hasWriteAccessToAction(string $action, ?User $user = null): bool
{
    return $this->getActionAccessControl()->hasWriteAccess(static::class, $action, $user);
}

protected function hasWriteAccessToControllerAction(string $controller, string $action, ?User $user = null): bool
{
    return $this->getActionAccessControl()->hasWriteAccess($controller, $action, $user);
}
```
---
#### Solution 3: Create a new method that will expose a controller and use both of them
```php
protected function hasWriteAccessToAction(string $action, ?User $user = null): bool
{
    return $this->getActionAccessControl()->hasWriteAccess(static::class, $action, $user);
}

protected function hasWriteAccessToControllerAction(string $controller, string $action, ?User $user = null): bool
{
    return $this->getActionAccessControl()->hasWriteAccess($controller, $action, $user);
}
```

Please, vote (if it is possible 😅):
| User | Solution 1 | Solution 2 | Solution 3 | Other solution |
| --- | :---: | :---: | :---: | :---: |
| @gabrielfs7  | <ul><li> - [ ] </li></ul> | <ul><li> - [ ] </li></ul> | <ul><li> - [ ] </li></ul> | <ul><li> - [ ] </li></ul> |
| @boajer  | <ul><li> - [ ] </li></ul> | <ul><li> - [ ] </li></ul> | <ul><li> - [ ] </li></ul> | <ul><li> - [ ] </li></ul> |
| @chinnu-focaloid  | <ul><li> - [ ] </li></ul> | <ul><li> - [ ] </li></ul> | <ul><li> - [ ] </li></ul> | <ul><li> - [ ] </li></ul> |

Comment (gabrielfs7) [reply actions/class.CommonModule.php:109]: Another solution is also have this method receiving an array of parameters or a value object, so next time we need a new parameter we do not need to break the method signature :)

**Review by gabrielfs7 — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

File: actions/class.CommonModule.php (hunk 1 above)
Comment (boajer) [reply actions/class.CommonModule.php:109]: I would vote for the value object if we still have time for those changes

**Review by chinnu-focaloid — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [X] New code is respecting code style rules
- [X] New code is respecting best practices
- [X] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [X] Acceptance criteria are respected
- [X] Pull request title and description are meaningful

**Review by boajer — APPROVED** [review]
LFGM, especially when we reach agreement on the deprecation 
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Comment by AntonNeviarouski** [comment]
QA verified on http://test-shpran.playground.kitchen.it.taocloud.org:41481/

**Pull request merged**